src/Core/UserImport.php

```diff
@@ -0,0 +1,301 @@
+	{
+		if (isset($arr['id'])) {
+			unset($arr['id']);
+		}
+
+		self::check_cols($table, $arr);
+		$cols = implode("`,`", array_map('dbesc', array_keys($arr)));
+		$vals = implode("','", array_map('dbesc', array_values($arr)));
+		$query = "INSERT INTO `$table` (`$cols`) VALUES ('$vals')";
+		logger("uimport: $query", LOGGER_TRACE);
+
+		if (IMPORT_DEBUG) {
+			return true;
+		}
+
+		return q($query);
+	}
+
+	/**
+	 * @brief Import account file exported from mod/uexport
+	 *
+	 * @param App $a Friendica App Class
+	 * @param array $file array from $_FILES
+	 */
+	public static function importAccount(App $a, $file)
+	{
+		logger("Start user import from " . $file['tmp_name']);
+		/*
+		STEPS
+		1. checks
+		2. replace old baseurl with new baseurl
+		3. import data (look at user id and contacts id)
+		4. archive non-dfrn contacts
+		5. send message to dfrn contacts
+		*/
+
+		$account = json_decode(file_get_contents($file['tmp_name']), true);
+		if ($account === null) {
+			notice(t("Error decoding account file"));
+			return;
+		}
+
+
+		if (!x($account, 'version')) {
+			notice(t("Error! No version data in file! This is not a Friendica account file?"));
+			return;
+		}
+
+		// check for username
+		$r = dba::selectFirst('user', ['uid'], ['nickname' => $account['user']['nickname']]);
+		if ($r === false) {
+			logger("uimport:check nickname : ERROR : " . dba::errorMessage(), LOGGER_NORMAL);
+			notice(t('Error! Cannot check nickname'));
+			return;
+		}
+
+		if (DBM::is_result($r) > 0) {
+			notice(sprintf(t("User '%s' already exists on this server!"), $account['user']['nickname']));
+			return;
+		}
+
+		// check if username matches deleted account
+		$r = dba::selectFirst('userd', ['id'], ['username' => $account['user']['nickname']]);
+		if ($r === false) {
+			logger("uimport:check nickname : ERROR : " . dba::errorMessage(), LOGGER_NORMAL);
+			notice(t('Error! Cannot check nickname'));
+			return;
+		}
+
+		if (DBM::is_result($r) > 0) {
+			notice(sprintf(t("User '%s' already exists on this server!"), $account['user']['nickname']));
+			return;
+		}
+
+		$oldbaseurl = $account['baseurl'];
+		$newbaseurl = System::baseUrl();
+
+		$oldaddr = str_replace('http://', '@', normalise_link($oldbaseurl));
+		$newaddr = str_replace('http://', '@', normalise_link($newbaseurl));
+
+		if (!empty($account['profile']['addr'])) {
+			$old_handle = $account['profile']['addr'];
+		} else {
+			$old_handle = $account['user']['nickname'].$oldaddr;
+		}
+
+		$olduid = $account['user']['uid'];
+
+		unset($account['user']['uid']);
+		unset($account['user']['account_expired']);
+		unset($account['user']['account_expires_on']);
+		unset($account['user']['expire_notification_sent']);
+
+		$callback = function ($key, $value) use ($oldbaseurl, $oldaddr, $newbaseurl, $newaddr) {
+			return str_replace(array($oldbaseurl, $oldaddr), array($newbaseurl, $newaddr), $value);
+		};
+
+		$v = array_map($account['user'], $callback);
```

Comment:
Did you check this works in a closed test environment like phpFiddle?

Reply:
The correct syntax is the following:
````php
$callback = function (&$value) use ($oldbaseurl, $oldaddr, $newbaseurl, $newaddr) {
    $value = str_replace(array($oldbaseurl, $oldaddr), array($newbaseurl, $newaddr), $value);
};

array_walk($account['user'], $callback);
````
`array_map` returns a new array while `array_walk` modifies the passed array.